python/ray/train/v2/_internal/execution/callback.py

```diff
@@ -10,15 +10,16 @@
     from ray.train.v2._internal.execution.failure_handling import FailureDecision
     from ray.train.v2._internal.execution.scaling_policy import ScalingDecision
     from ray.train.v2._internal.execution.worker_group import (
+        Worker,
         WorkerGroup,
-        WorkerGroupStatus,
+        WorkerGroupPollStatus,
     )
 
 
 @DeveloperAPI
 class WorkerGroupCallback(RayTrainCallback):
     def before_init_train_context(
-        self, worker_group: "WorkerGroup"
+        self, workers: List["Worker"]
```

Comment:
Why change this to workers?

Reply:
This is because the `WorkerGroup` itself is still in the middle of its own creation stage. 

Calling `WorkerGroup.execute` would fail here because the `WorkerGroup` is not yet "active" when calling this step. So I wanted to constrain this callback hook to what is "ready" which is the workers.

Reply:
Got it, this callback is the only one that runs into that problem.